astroquery/cadc/core.py

```diff
@@ -237,6 +239,136 @@ def get_collections(self):
+
+        if get_query_payload:
+            return request_payload
+
+        query = request_payload['query'] + " AND (dataProductType = 'image')"
+        response = self.run_query(query, operation='sync')
+        query_result = response.get_results()
+
+        if query_result and len(query_result) == 2000:
+            logger.debug("Synchronous query results capped at 2000 results - results may be truncated")
+
+        images_urls = self.get_image_list(query_result, coordinates, radius)
+        images = []
+
+        try:
+            readable_objects = [commons.FileContainer(url, encoding='binary',
+                                                      show_progress=show_progress) for url in images_urls]
+            for obj in readable_objects:
+                images.append(obj.get_fits())
+        except HTTPError as err:
+            logger.debug(
+                "{} - Problem retrieving the file: {}".format(str(err), str(err.url)))
+            pass
+
+        return images
+
+    @class_or_instance
+    def get_image_list(self, query_result, coordinates, radius):
+        """
+        Function to map the results of a CADC query into URLs to
+        corresponding data and cutouts that can be later downloaded.
+
+        The function uses the IVOA DataLink Service
+        (http://www.ivoa.net/documents/DataLink/) implemented at the CADC.
+        It works directly with the results produced by Cadc.query_region and
+        Cadc.query_name but in principle it can work with other query
+        results produced with the Cadc query as long as the results
+        contain the 'caomPublisherID' column. This column is part of the
+        caom2.Plane table.
+
+        Parameters
+        ----------
+        query_result : result returned by Cadc.query_region() or
+                    Cadc.query_name(). In general, the result of any
+                    CADC TAP query that contains the 'caomPublisherID' column
+                    can be use here.
+        coordinates : str or `astropy.coordinates`.
+            Coordinates around which to query
+        radius : str or `astropy.units.Quantity`.
+            The radius of the cone search AND cutout area
+
+        Returns
+        -------
+        A list of URLs to data.
+        """
+
+        def chunks(obj_list, chunk_len):
+            """
+            A generator that breaks list obj_list into sublists of length chunk_len
+            :param obj_list: The list to be chunked
+            :param chunk_len: The length of each chunked sublist
+            :return: An iterator that goes through each sublist
+            """
+            for idx in range(0, len(obj_list), chunk_len):
+                yield obj_list[idx:idx + chunk_len]
+
+        if not query_result:
+            raise AttributeError('Missing query_result argument')
+
+        # Send datalink requests in batches of 20 publisher ids
+        n_pids = 20
```

Comment:
Please move closer to its usage.

Reply:
sure, i'll change the name to `batch_size` as well